crud/select/executor.lua

```diff
@@ -84,17 +84,26 @@ function executor.execute(space, index, filter_func, opts)
     end
 
     local tuple
-    local gen = index:pairs(value, {iterator = opts.tarantool_iter})
+    local raw_gen = index:pairs(value, {iterator = opts.tarantool_iter})
+    local gen = fun.wrap(function(param, state)
+        local next_state, var = raw_gen(param, state)
+
+        if var ~= nil then
+            resp.tuples_lookup = resp.tuples_lookup + 1
+        end
+
+        return next_state, var
+    end, raw_gen.param, raw_gen.state)
```

Comment:
A content of the luafun iterator table is not documented and actually the implementation details. It is better to do `local raw_gen, param, state = index:pairs(<...>)`.

Reply:
BTW, are we need `fun.wrap()` here? It seems, no.

Reply:
Yes, we need: `scroll_to_after_tuple` uses `gen.param, gen.state`

Reply:
Not nice, but not the scope of this issue. Okay so.